src/calendar/calendar.tsx

```diff
@@ -54,6 +54,8 @@ export const Component = (
     const [selectedStartDate, setSelectedStartDate] = useState<string>(); // YYYY-MM-DD
     const [selectedEndDate, setSelectedEndDate] = useState<string>(); // YYYY-MM-DD
 
+    const [isNewSelection, setIsNewSelection] = useState<boolean>(false);
```

Comment:
can trouble you to flip the boolean values around for clarity

isNewSelection = true to mean user changed the selection in the calendar
isNewSelection = false to mean it's an existing selection before the calendar was open